letsencrypt/client/display/ops.py

```diff
@@ -6,41 +7,103 @@
 from letsencrypt.client import interfaces
 from letsencrypt.client.display import util as display_util
 
+
 # Define a helper function to avoid verbose code
 util = zope.component.getUtility  # pylint: disable=invalid-name
 
 
-def choose_authenticator(auths, errs):
-    """Allow the user to choose their authenticator.
+def choose_plugin(prepared, question):
+    """Allow the user to choose ther plugin.
 
-    :param list auths: Where each of type
-        :class:`letsencrypt.client.interfaces.IAuthenticator` object
-    :param dict errs: Mapping IAuthenticator objects to error messages
+    :param list prepared: List of `~.PluginEntryPoint`.
+    :param str question: Question to be presented to the user.
 
-    :returns: Authenticator selected
-    :rtype: :class:`letsencrypt.client.interfaces.IAuthenticator` or `None`
+    :returns: Plugin entry point chosen by the user.
+    :rtype: `~.PluginEntryPoint`
 
     """
-    descs = [auth.description if auth not in errs
-             else "%s (Misconfigured)" % auth.description
-             for auth in auths]
+    opts = [plugin_ep.name_with_description
```

Comment:
Description with name would be much better as the description was always meant to be observed by the end user.  I do agree that the plugin name should be readily observable given the help output references.

I do not believe we should be shipping with multiple redundant plugins.  If a developer wants to create their own, that is fine and they should be able to figure out what is going on and write an accurate description. If we have to rely on the uniqueness of the plugin name in our packaged product, we have failed.